src/RestApi/StoreApi/Schemas/CartShippingRateSchema.php

```diff
@@ -0,0 +1,242 @@
+
+defined( 'ABSPATH' ) || exit;
+
+use \WC_Shipping_Rate as ShippingRate;
+
+/**
+ * CartShippingRateSchema class.
+ */
+class CartShippingRateSchema extends AbstractSchema {
+	/**
+	 * The schema item name.
+	 *
+	 * @var string
+	 */
+	protected $title = 'cart-shipping-rate';
+
+	/**
+	 * Cart schema properties.
+	 *
+	 * @return array
+	 */
+	protected function get_properties() {
+		return [
+			'destination'    => [
+				'description' => __( 'Shipping destination address.', 'woo-gutenberg-products-block' ),
+				'type'        => 'object',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+				'items'       => [
+					'address_1' => [
+						'description' => __( 'First line of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+					'address_2' => [
+						'description' => __( 'Second line of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+					'city'      => [
+						'description' => __( 'City of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+					'state'     => [
+						'description' => __( 'ISO code, or name, for the state, province, or district of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+					'postcode'  => [
+						'description' => __( 'Zip or Postcode of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+					'country'   => [
+						'description' => __( 'ISO code for the country of the address being shipped to.', 'woo-gutenberg-products-block' ),
+						'type'        => 'string',
+						'context'     => [ 'view', 'edit' ],
+						'readonly'    => true,
+					],
+				],
+			],
+			'items'          => [
+				'description' => __( 'List of cart items (keys) the returned shipping rates apply to.', 'woo-gutenberg-products-block' ),
+				'type'        => 'array',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+				'items'       => [
+					'type' => 'string',
+				],
+			],
+			'shipping-rates' => [
+				'description' => __( 'List of shipping rates.', 'woo-gutenberg-products-block' ),
+				'type'        => 'array',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+				'items'       => $this->get_rate_properties(),
+			],
+		];
+	}
+
+	/**
+	 * Schema for a single rate.
+	 *
+	 * @return array
+	 */
+	protected function get_rate_properties() {
+		return [
```

Comment:
I think the schema needs to have an additional level to include the type for each item in the array?  In other words:

```php
return [
     'type' => 'object',
     'properties' => [
         // current shipping rate properties you have ( `name`, `description` etc)
     ]
```